src/Block/AdminSearchBlockService.php

```diff
@@ -68,6 +68,14 @@ public function execute(BlockContextInterface $blockContext, Response $response
             $blockContext->getSetting('per_page')
         );
 
+        if (false === $pager) {
+            if (null === $response) {
+                $response = new Response();
+            }
+
+            return $response->setContent('')->setStatusCode(204);
```

Comment:
you can just do `new Response('', 204)`

Reply:
ah, but this is set for every `$resposne` where the `$pager` is false? So we could have a $response but no $pager?

Reply:
Yeah, a $response might be given through the parameter of the function. If it isn't (when it's null), one is created afterwards. Hence the extra checks.

Reply:
then as you wrote it should be ok.

Reply:
If you are setting content and status code, why are you reusing the request, what could come with the original one?  Cookies? Custom headers?

Reply:
The TTL is set from a BlockContext in BlockRenderer::render().